src/qibocal/protocols/readout_optimization/resonator_optimization.py

```diff
@@ -0,0 +1,330 @@
+from dataclasses import dataclass, field
+
+import numpy as np
+import numpy.typing as npt
+import plotly.graph_objects as go
+import scipy.ndimage as ndimage
+from plotly.subplots import make_subplots
+from qibolab import (
+    AcquisitionType,
+    Delay,
+    Parameter,
+    PulseSequence,
+    Sweeper,
+)
+
+from qibocal import update
+from qibocal.auto.operation import Data, Parameters, QubitId, Results, Routine
+from qibocal.calibration import CalibrationPlatform
+from qibocal.fitting.classifier.qubit_fit import QubitFit
+from qibocal.protocols.utils import HZ_TO_GHZ, readout_frequency, table_dict, table_html
+
+__all__ = ["resonator_optimization"]
+
+
+@dataclass
+class ResonatorOptimizationParameters(Parameters):
+    """Resonator optimization runcard inputs"""
+
+    freq_width: int
+    """Width for frequency sweep relative  to the readout frequency [Hz]."""
+    freq_step: int
+    """Frequency step for sweep [Hz]."""
+    amplitude_start: float
+    """Minimum amplitude."""
+    amplitude_stop: float
+    """Maximum amplitude."""
```

Comment:
```suggestion
    amplitude_min: float
    """Minimum amplitude."""
    amplitude_max: float
    """Maximum amplitude."""
```
Since amplitude is not a time related quantity maybe using `min` and `max` is better in my opinion.